### PR TITLE
[MRG+1] Fix writing of raws with empty annotations.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/4164

Now the writer checks if the annotations are empty and doesn't try to write it in that case. It was previously trying to set empty annotations to None, but this is probably more appropriate.

### CODE REVIEW

File: examples/io/plot_objects_from_arrays.py [hunk 1]
```diff
@@ -148,7 +148,7 @@
 for ai, asig in enumerate(seg.analogsignals):
     # Since the data does not contain channel names, channel indices are used.
     ch_names.append('Neo %02d' % (ai + 1,))
-    data.append(asig.rescale('V').magnitude[:, 0])
+    data.append(asig.rescale('V').magnitude)
```
Comment (jaeilepp) [inline examples/io/plot_objects_from_arrays.py]: Now also includes this fix for the circle build.

**Comment by agramfort** [comment]
please update what's new and we need to backport this. can you take care of it?

**Comment by jaeilepp** [comment]
You mean what's new on 0.14? There is an entry for 0.15.

File: examples/io/plot_objects_from_arrays.py (hunk 1 above)
Comment (Eric89GXL) [inline examples/io/plot_objects_from_arrays.py]: I already fixed this in `master` I think...

Comment (Eric89GXL) [inline examples/io/plot_objects_from_arrays.py]: ... in other words, this will *undo* my fix. I think the version I added works for Neo 0.5.0 (latest) and your change here will work for older versions (what we had before)

Comment (Eric89GXL) [inline examples/io/plot_objects_from_arrays.py]: I can confirm that with latest `neo` this breaks things:
```
Traceback (most recent call last):
  File "examples/io/plot_objects_from_arrays.py", line 159, in <module>
    raw = mne.io.RawArray(data, info)
  File "<string>", line 2, in __init__
  File "/home/larsoner/custombuilds/mne-python/mne/utils.py", line 707, in verbose
    return function(*args, **kwargs)
  File "/home/larsoner/custombuilds/mne-python/mne/io/array/array.py", line 59, in __init__
    raise ValueError('Data must be a 2D array of shape (n_channels, '
ValueError: Data must be a 2D array of shape (n_channels, n_samples
```
Please revert and we should deal with this in a separate PR.

Comment (jaeilepp) [inline examples/io/plot_objects_from_arrays.py]: Ok. It's probably ok to revert this change then.

**Comment by Eric89GXL** [comment]
This PR does not fix reading of already-broken files. I still get the same `TypeError: unsupported operand type(s) for -: 'NoneType' and 'NoneType'` error. Unfortunately we have written some files like this already, so we need to support reading them.

**Comment by Eric89GXL** [comment]
In other words, in #4164 I pointed out that we have two bugs in our code, and this PR currently only fixes one of them.

**Comment by jaeilepp** [comment]
This is ready for review.

**Comment by Eric89GXL** [comment]
I can read the old file now and the code LGTM, +1 for merge

**Comment by agramfort** [comment]
@jaeilepp please merge and backport.

thanks a lot

**Comment by kambysese** [comment]
@jaeilepp could you please ping me once this is merged into master. Thanks in advance.

**Comment by Eric89GXL** [comment]
Done. @jaeilepp you just need to backport now

**Comment by jaeilepp** [comment]
@kambysese backported.

**Comment by kambysese** [comment]
@jaeilepp thank you sir(s)
